docs/pluginconfig.md

```diff
@@ -0,0 +1,39 @@
+# Initiating Plugin - Versioning model
+
+When initiating plugin code and calling `GenerateConfig` method we need to
+pass in `Plugin Template/PluginConfig` as a parameter. 
+
+Plugin template defines main characteristics for the cluster. In example 
+cluster, images versions, VM image, external integration details.
+
+Config file examples can be found in `pluginconfig/pluginconfig-{version}.yaml`, but ideally config object should be constructed using `pkg/api/plugin/config.go Config` struct.
```

Comment:
a config file is used to simplify prod configuration change without new binary rollout, however in general the struct contains secrets and it may not be wanted to store these in a yaml file